app/gui2/src/components/GraphNode.vue

```diff
@@ -244,11 +248,19 @@ onUpdated(() => {
   }
 })
 
-function handleClick(e: PointerEvent) {
-  if (e.shiftKey) {
-    emit('delete')
-    e.preventDefault()
-    e.stopPropagation()
+// TODO [sb]: restore delete functionality (DELETE key)
+function handleClick(event: PointerEvent) {
+  if (shortcutRegistry.matchesMouseAction('replace-nodes-selection', event)) {
+    emit('replaceSelection')
+  } else if (shortcutRegistry.matchesMouseAction('add-to-nodes-selection', event)) {
+    emit('update:selected', true)
+  } else if (shortcutRegistry.matchesMouseAction('remove-from-nodes-selection', event)) {
+    emit('update:selected', false)
+  } else if (
+    shortcutRegistry.matchesMouseAction('toggle-nodes-selection', event) ||
+    shortcutRegistry.matchesMouseAction('invert-nodes-selection', event)
+  ) {
+    emit('update:selected', !props.selected)
```

Comment:
The whole shortcut management looks messy and highly repetitive right now. I think it's worth giving it another shot with a better API in mind. Something like this could be a nice goal:

```ts
// Definition of keybinds namespace. Executed during module load (top level of script).
// The name (first argument) is used only for config matching purposes.
const graphBindings = keybinds.defineNamespace('graph-editor', {
    // `Mod` is a custom modifier, matching either `Ctrl` on Windows or `Command` on Mac
    undo: ['Mod+Z'],
    // Multiple alternative shortcuts can be specified, both work the same way.
    redo: ['Mod+Y', 'Mod+Shift+Z'],
    // Allows binding to mouse (pointer) buttons, accepts modifier keys with them.
    dragScene: ['PointerAux', 'Ctrl+PointerMain'],
    // ...more actions. The order here doesn't matter, but names do.
})
  
  
// 'keydown' event handler of a component:
function onKeyDown(event: KeyboardEvent) {
    graphBindings.handle(event, {
        // The order of specified object keys determines the order of matching. Only the first
        // matching action is executed. Once an action is matched, the event automatically stops
        // propagating and default behavior is prevented.
        undo() {
            // Undo action matched here
        },
        redo() {
            // Redo action matched here
        },
    
        [keybinds.DefaultHandler]() {
            // Optionally, when no action was matched in this block, do something else
        },
    })
}

// 'mousedown' event handler in a component. Should also work the same with pointer events.
function onKeyDown(event: MouseEvent) {
    // perform singular match. Equivalent to `handle` call with a single action, returns `true` if
    // the event is matching, automatically stops propagation and prevents default behavior.
    if (graphBindings.match(event, 'dragScene')) {

    }
}
```

Note, we could totally integrate that into the `usePointer` composable, so it accepts the event name to react to.